### PR TITLE
85 add testcases to server gamecontroller

### PR DESCRIPTION
**Pull request by serainaschraff**
finally done :)

100% code coverage for game controller and minor bug fixes / other tests

### CODE REVIEW

**Comment by serainaschraff** [comment]
Check failed because of a bug in a different test:

GameServiceTest > getNRandomBlackCards() FAILED
    java.lang.NullPointerException: Cannot invoke "org.springframework.data.domain.Page.getTotalElements()" because "somePage" is null
        at ch.uzh.ifi.hase.soprafs22.service.GameService.getNRandomBlackCards(GameService.java:63)
        at ch.uzh.ifi.hase.soprafs22.service.GameServiceTest.getNRandomBlackCards(GameServiceTest.java:79)

**Comment by anhuwi** [comment]
So i will merge mine in the dev and den merge the dev again in yours?
If  i solved the issue, there will be no longer a problem?

**Comment by serainaschraff** [comment]
> So i will merge mine in the dev and den merge the dev again in yours? If i solved the issue, there will be no longer a problem?

I as the error is in a file I have not atered but you have and I checked your code, the test causing the fail of the build does no longer exist in the code, I assume it will be fine as all my tests run properly on my local machine, and also online only one test failed..

File: src/main/java/ch/uzh/ifi/hase/soprafs22/controller/GameController.java [hunk 1]
```diff
@@ -101,7 +101,7 @@ public void createGame(@RequestHeader(value = "authorization", required = false)
   }
 
     @GetMapping("users/{userId}/games/blackCards")
-    @ResponseStatus(HttpStatus.CREATED)
+    @ResponseStatus(HttpStatus.OK)
```
Comment (anhuwi) [inline src/main/java/ch/uzh/ifi/hase/soprafs22/controller/GameController.java:104]: Is this in line with the api excel?

Comment (serainaschraff) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/controller/GameController.java:104]: a get should never create anything.. and yes the excel states code 200

**Pull request merged**